driver-core/src/test/functional/com/mongodb/client/model/expressions/TypeExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import com.mongodb.MongoCommandException;
+import org.bson.BsonDocument;
+import org.bson.types.Decimal128;
+import org.bson.types.ObjectId;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+
+import static com.mongodb.client.model.expressions.Expressions.of;
+import static com.mongodb.client.model.expressions.Expressions.ofIntegerArray;
+import static com.mongodb.client.model.expressions.Expressions.ofNull;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE_TIME;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class TypeExpressionsFunctionalTest extends AbstractExpressionsFunctionalTest {
+    // https://www.mongodb.com/docs/manual/reference/operator/aggregation/#type-expression-operators
+
+    // https://www.mongodb.com/docs/manual/reference/operator/aggregation/type/ (28 |40)
```

Comment:
The numbers are the raking by total operations and by group. The goal is to get the top-50 in. I think the comments are useful for reference, while the numbers useful for me and otherwise harmless (I don't know where to doc their meaning).

Reply:
Without explaining in the comments what the numbers are, they are indecipherable to everyone but the author. Though specifically the ranking information is not useful as comments in code either way. I disagree with the commenting approach taken in the expressions PRs. The code filled with comment pieces like this one, with comments that appear to represent draft notes on reasoning behind some decisions (see [here](https://github.com/mongodb/mongo-java-driver/pull/1050#discussion_r1031125541)), etc.

If others (@jyemin, @rozza) are fine with this approach to code comments, I will push this further. I disagree with the approach specifically because:

- Comments left in code should be helpful to future readers and should be written clearly (if they are draft notes, they should not be left in the code).
- Comments should be put on/close to relevant program elements (as comments or potentially even as javadocs if the program elements are not public API). Readers will not look for thoughts behind the API design in the test code.

Reply:
Please remove the numbers.  They are just going to confuse people